viewing_party/party.py

```diff
@@ -1,23 +1,192 @@
 # ------------- WAVE 1 --------------------
+def create_movie(title: str, genre: str, rating: float):
+    """ Create a movie dictionary utilizing the arguments and return it.
+    If one of the arguments is Falsy, instead return None.
 
-def create_movie(title, genre, rating):
-    pass
+    Keyword arguments:
+    title -- a string movie title
+    genre -- a string genre name
+    rating -- a float value
+    """
+    if title and genre and rating:
+        movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data: dict, movie: dict):
+    """ Add a movie dictionary to the "watched" key in user_data.
+    Return user_data.
+
+    Keyword arguments:
+    user_data -- a dictionary with a "watched" key with a list value
+    movie -- a dictionary with movie info
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data: dict, movie: dict):
+    """ Add a movie dictionary to the "watchlist" key in user_data.
+    Return user_data.
+
+    Keyword arguments:
+    user_data -- a dictionary with a "watchlist" key with a list value
+    movie -- a diictionary with movie info
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data: dict, title: str):
+    """ Move a movie in user_data from "watchlist" to "watched".
+    Return user_data.
+
+    Keyword arguments:
+    user_data -- a dictionary with "watchlist" and "watched" keys
+    title -- the title of the movie being moved
+    """
+    counter = 0
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            movie_watched = user_data["watchlist"].pop(counter)

```

Comment:
This is a totally good use of `pop` while keeping track of `counter` but one suggestion is that keeping mental track of counter is an additional step that may not be needed. If you use `remove` with the current `movie`, it's safe because you know for sure that `movie` is in the `user_data["watchlist"]`. So you could replace line 54 with
```python
user_data["watchlist"].remove(movie)
```
then since that same `movie` is the one you want to `append` to `user_data["watched"]`, you can just do
```python
user_data["watched"].append(movie)
```